src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java

```diff
@@ -119,6 +126,20 @@ public ProbationRecord getProbationRecord(String crn, boolean applyDocumentFilte
         return probationRecord;
     }
 
+    private Mono<Conviction> enrichConviction(String crn, Conviction conviction) {
+
+        var convictionId = Long.valueOf(conviction.getConvictionId());
+        var enrichedConvictionMono = Mono.zip(
```

Comment:
We only need to retrieve this lot for active convictions on the Probation Record screen I think, it'll show all the requirements on the order screen if they click through but for this page we only show them anything current.

Reply:
Yes - that's true and I asked Paul about that. I did wonder about CCS though generally. So in this area, it feels like we have an endpoint which is very specific for prepare a case (which it is :-) ) It returns active and inactive conviction requirements , only active PSS requirements and licence conditions. That starts to feel muddled. It's all so hardcoded to prepare a case. I started to feel uncomfortable.If we were doing something general, it would be active and inactive and let the client decide what's needed.

Reply:
Can do it, I just think we need to consider this for the future. We ought to document the responses around active / inactive so it's clear

Reply:
Yeah I know what you mean,  court-case-service is a bit confused between pure API and back-end for a front-end. Given our current use case is purely to serve the prepare-a-case UI I think that's probably ok for now, we ought not to let any other services call the probation record endpoint in particular though. If we do get any more consumers I think a bit of  ringfencing would be needed at the very least. 

FWIW the data *owned* by the court-case-service is pretty clean in the API it serves up. In that sense we've kinda got 2 different services rolled into 1.